python/cuml/dask/ensemble/randomforestcommon.py

```diff
@@ -0,0 +1,173 @@
+        the treelite format and then concatenate the different treelite models
+        to create a single model. The concatenated model is then converted to
+        bytes format.
+        """
+        mod_bytes = []
+        for w in self.workers:
+            mod_bytes.append(self.rfs[w].result().model_pbuf_bytes)
+        last_worker = w
+        all_tl_mod_handles = []
+        model = self.rfs[last_worker].result()
+        for n in range(len(self.workers)):
+            all_tl_mod_handles.append(model._tl_model_handles(mod_bytes[n]))
+        concat_model_handle = model._concatenate_treelite_handle(
+            treelite_handle=all_tl_mod_handles)
+
+        model._concatenate_model_bytes(concat_model_handle)
+        return model
+
+    def _predict_using_fil(self, X, delayed, **kwargs):
+        data = DistributedDataHandler.create(X, client=self.client)
+        self.datatype = data.datatype
+        return self._predict(X, delayed=delayed, **kwargs)
+
+    def _get_params(self, deep):
+        model_params = list()
+        for n, worker in enumerate(self.workers):
+            model_params.append(
+                self.client.submit(
+                    _func_get_params,
+                    self.rfs[worker],
+                    deep,
+                    workers=[worker]
+                )
+            )
+        wait(model_params)
+        raise_exception_from_futures(model_params)
+        params_of_each_model = list()
+        for i in range(len(model_params)):
+            params_of_each_model.append(model_params[i].result())
+        return params_of_each_model
+
+    def _set_params(self, worker_numb, **params):
+        model_params = list()
+        if worker_numb is None:
+            worker_numb = [posi for posi in range(len(self.workers))]
+        workers = [worker for worker in self.workers]
+        for i in worker_numb:
+            model_params.append(
+                self.client.submit(
+                    _func_set_params,
+                    self.rfs[workers[i]],
+                    **params,
+                    workers=[workers[i]]
+                )
+            )
+        wait(model_params)
+        raise_exception_from_futures(model_params)
+        return self
+
+    def _print_summary(self):
+        """
+        Print the summary of the forest used to train and test the model.
+        """
+        futures = list()
+        for n, w in enumerate(self.workers):
+            futures.append(
+                self.client.submit(
+                    _print_summary_func,
+                    self.rfs[w],
+                    workers=[w],
+                )
+            )
+
+        wait(futures)
+        raise_exception_from_futures(futures)
+        return self
+
+
+def _func_fit(model, input_data, convert_dtype):
+    X = concatenate([item[0] for item in input_data])
+    y = concatenate([item[1] for item in input_data])
+    return model.fit(X, y, convert_dtype)
+
+
+def _print_summary_func(model):

```

Comment:
I'm not sure why we need these static functions? They look equivalent to the existing functions like model.print_summary. model.print_summary takes self and this one takes model, but otherwise they are the same.

Reply:
This is required to access the function `print_summary` present in the non dask RF cython code. 
The model passed here is the non dask RF model obtained from `self.rfs`.